src/alchemlyb/tests/parsing/test_gmx.py

```diff
@@ -103,3 +106,65 @@ def test_dHdl_case3():
 
             assert dHdl.index.names == ['time', 'fep-lambda', 'coul-lambda', 'vdw-lambda', 'restraint-lambda']
             assert dHdl.shape == (2500, 4)
+
+def test_u_nk_with_total_energy():
+    """Test that the reduced potential is calculated correctly when the total energy is given.
+
+    """
+
+    # Load dataset
+    dataset = load_water_particle_with_total_energy()
+
+    # Check if the sum of values on the diagonal has the correct value
+    assert _diag_sum(dataset) == 47611377946.58586
+
+    # Check one specific value in the dataframe
+    assert float(extract_u_nk(dataset['data']['AllStates'][0], T=300).iloc[0][0]) == -11250.643602869592
+
+def test_u_nk_with_potential_energy():
+    """Test that the reduced potential is calculated correctly when the potential energy is given.
+
+    """
+
+    # Load dataset
+    dataset = load_water_particle_with_potential_energy()
+
+    # Check if the sum of values on the diagonal has the correct value
+    assert _diag_sum(dataset) == 16674041445589.646
+
+    # Check one specific value in the dataframe
+    assert float(extract_u_nk(dataset['data']['AllStates'][0], T=300).iloc[0][0]) == -15659.655560881085
```

Comment:
needs to use `assert_almost_equal` or similar. You cannot compare floats for equality because this depends on hardware etc.